### PR TITLE
Separate instruction to install OS dependencies in images

### PR DESCRIPTION
**Pull request by potiuk**
This change will allow to experiment with other base images
(for example CentOS that our users highly demand) but also
it has a few nice simplifications and improvements along
the way:

* no more runtime parameters for CI image (they only make
  sense for PROD image)
* no more support for Buster image (it is end of life in August)
* Dockerfile has now less embedded default values (most of them
  moved to inlined bash script)
* configuration for yarn sources is removed (we do not need yarn
  any more in our images)
* additional pure-dev dependencies in CI image are passed through
  ADDITIONAL_DEV_DEPS
* dev installation does not remove installation cache, making
  the CI image slightly bigger but easier for devel use - to install new
  dependencies (no need for `apt-get update` before installation)
* latest patchlevels of various tools we use for CI were bumped

Based on top of https://github.com/apache/airflow/pull/25582 - please change only last commit.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
This is a refactoring for installation of `apt' dependencies in our Dockerfile and it will be really useful to explore a possibility of building a CentOS image which has been proposed by the Snowflake team in https://github.com/apache/airflow/pull/24825 

cc: @mik-laj @sfc-gh-mkmak - it's not yet adding CentOS but it should make it easy for us to be able to choose Debian/CentOS when building the image. Once this one is merged, I will attempt to incorporate what you've done at Snowflake and when I succceed I will make a propsal at the devlist to introduce an experimental CentOS support (once I know if we can easily have 1-1 parity for our PROD images).

**Comment by potiuk** [comment]
I need to get https://github.com/apache/airflow/pull/25567 to main and rebase for the CI to succeed (We are overiding default parameters in the CI image in the main version of breeze.

File: docs/docker-stack/build-arg-ref.rst [hunk 1]
```diff
@@ -130,33 +130,33 @@ for examples of using those arguments.
...
 +------------------------------------------+------------------------------------------+------------------------------------------+
 | ``ADDITIONAL_DEV_APT_COMMAND``           |                                          | Additional Dev apt command executed      |
 |                                          |                                          | before dev dep are installed             |
 |                                          |                                          | in the Build image. Should start with    |
 |                                          |                                          | ``&&``.                                  |
 +------------------------------------------+------------------------------------------+------------------------------------------+
-| ``DEV_APT_DEPS``                         | (see Dockerfile)                         | Dev APT dependencies installed           |
-|                                          |                                          | in the Build image.                      |
+| ``DEV_APT_DEPS``                         | Empty - install default de dependencies  | Dev APT dependencies installed           |
```
Comment (mik-laj) [inline docs/docker-stack/build-arg-ref.rst]: de?

File: docs/docker-stack/build-arg-ref.rst [hunk 2]
```diff
@@ -130,33 +130,33 @@ for examples of using those arguments.
...
 | ``ADDITIONAL_RUNTIME_APT_COMMAND``       |                                          | Additional Runtime apt command executed  |
 |                                          |                                          | before runtime dep are installed         |
-|                                          |                                          | in the Main image. Should start with     |
+|                                          |                                          | in the ``main`` stage. Should start with |
 |                                          |                                          | ``&&``.                                  |
 +------------------------------------------+------------------------------------------+------------------------------------------+
-| ``RUNTIME_APT_DEPS``                     | (see Dockerfile)                         | Runtime APT dependencies installed       |
-|                                          |                                          | in the Main image.                       |
+| ``RUNTIME_APT_DEPS``                     | Empty - install default de dependencies  | Runtime APT dependencies installed       |
```
Comment (mik-laj) [inline docs/docker-stack/build-arg-ref.rst]: de?

File: tests/charts/helm_template_generator.py [hunk 3]
```diff
@@ -102,7 +102,7 @@ def validate_k8s_object(instance, kubernetes_version):
 
 
 def render_chart(
-    name="RELEASE-NAME",
+    name="release-name",
```
Comment (mik-laj) [inline tests/charts/helm_template_generator.py]: Can you tell little more about it?

**Comment by mik-laj** [comment]
CC: @sfc-gh-mkmak

File: tests/charts/helm_template_generator.py (hunk 3 above)
Comment (potiuk) [reply tests/charts/helm_template_generator.py]: This based on previous PR https://github.com/apache/airflow/pull/25582. I rebased in on top to see if it works.

**Comment by potiuk** [comment]
More comments needed :) ?

**Comment by potiuk** [comment]
Anyone :) ?

**Comment by potiuk** [comment]
@mik-laj ?

**Review by mik-laj — APPROVED** [review]

**Pull request merged**